cgt_calc/parsers/freetrade.py

```diff
@@ -0,0 +1,208 @@
+                amount /= fx_rate
+            currency = "GBP"
+        elif action == ActionType.DIVIDEND:
+            # Total amount before US tax withholding
+            amount = Decimal(row["Dividend Gross Distribution Amount"])
+            quantity, price = None, None
+            currency = row["Instrument Currency"]
+            if currency != "GBP":
+                # FX Rate is not defined for dividends,
+                # but we can use base one as there's no fee
+                amount /= Decimal(row["Base FX Rate"])
+            currency = "GBP"
+        elif action in [ActionType.TRANSFER, ActionType.INTEREST]:
+            amount = Decimal(row["Total Amount"])
+            quantity, price = None, None
+            currency = "GBP"
+        else:
+            raise ParsingError(
+                filename, f"Numbers parsing unimplemented for action: {action}"
+            )
+
+        if row["Type"] == "FREESHARE_ORDER":
+            price = Decimal(0)
+            amount = Decimal(0)
+
+        amount_negative = action == ActionType.BUY or row["Type"] == "WITHDRAWAL"
+        if amount is not None and amount_negative:
+            amount *= -1
+
+        super().__init__(
+            date=parse_date(row["Timestamp"]),
+            action=action,
+            symbol=symbol,
+            description=f"{row['Title']} {action}",
+            quantity=quantity,
+            price=price,
+            fees=Decimal(0),  # not implemented
+            amount=amount,
+            currency=currency,
+            broker="Freetrade",
+        )
+
+
+def action_from_str(type: str, buy_sell: str, filename: str) -> ActionType:
+    """Infer action type."""
+    if type == "INTEREST_FROM_CASH":
+        return ActionType.INTEREST
+    if type == "DIVIDEND":
+        return ActionType.DIVIDEND
+    if type in ["TOP_UP", "WITHDRAWAL"]:
+        return ActionType.TRANSFER
+    if type in ["ORDER", "FREESHARE_ORDER"]:
+        if buy_sell == "BUY":
+            return ActionType.BUY
+        if buy_sell == "SELL":
+            return ActionType.SELL
+
+        raise ParsingError(filename, f"Unknown buy_sell: {buy_sell}")
+
+    raise ParsingError(filename, f"Unknown type: {type}")
+
+
+def validate_header(header: list[str], filename: str) -> None:
+    """Check if header is valid."""
+    for actual in header:
+        if actual not in COLUMNS:
+            msg = f"Unknown column {actual}"
+            raise ParsingError(filename, msg)
+
+
+@dataclass
+class StockSplit:
+    """Info about stock split."""
+
+    symbol: str
+    date: date
+    factor: int
+
+
+STOCK_SPLIT_INFO = [
+    StockSplit(symbol="GOOGL", date=datetime(2022, 7, 18).date(), factor=20),
+    StockSplit(symbol="TSLA", date=datetime(2022, 8, 25).date(), factor=3),
+    StockSplit(symbol="NDAQ", date=datetime(2022, 8, 29).date(), factor=3),
+]
+
+
+# Pretend there was no split
+def _handle_stock_split(transaction: BrokerTransaction) -> BrokerTransaction:
```

Comment:
Mmm why are you adding this? I think it would make more sense to handle splits in the main class , not in the parser

Reply:
Maybe you can keep this part separated from this PR since is fixing a different issue